src/js/components/Widgets/PositionSupportOpposeSnippet.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component, PropTypes } from "react";
+
+export default class PositionSupportOpposeSnippet extends Component {
+  static propTypes = {
+    candidate_display_name: PropTypes.string,
+    is_on_candidate_page: PropTypes.bool,
+    is_support: PropTypes.bool.isRequired,
+    is_oppose: PropTypes.bool.isRequired,
+    more_info_url: PropTypes.string,
+    speaker_display_name: PropTypes.string,
+    statement_text: PropTypes.string
+  };
+
+  render () {
+    var src;
+    var className;
+    var alt;
+    var positionLabel;
+    var isSupportedBy;
+    if (this.props.is_support){
+      src = "/img/global/icons/thumbs-up-color-icon.svg";
+      className = "explicit-position__icon";
+      alt = "Supports";
+      positionLabel = "Supports";
+      isSupportedBy = "is Supported by"
+    } else if (this.props.is_oppose) {
+      src = "/img/global/icons/thumbs-down-color-icon.svg";
+      className = "explicit-position__icon";
+      alt = "Opposes";
+      positionLabel = "Opposes";
+      isSupportedBy = "is Opposed by"
+    } else {
+      src = "/img/global/icons/mixed-rating-icon.svg";
+      className = "position-rating__icon position-rating__icon--mixed";
+      alt = "Neutral Rating";
+      positionLabel = "Neutral Rating of";
+      isSupportedBy = "is viewed Neutrally by"
```

Comment:
Hmm... Okay. Semantically, it still feels different than the (explicit)
position ratings. I understand that this might be a simpler way to execute,
but I find the grouping to be a bit confusing (FWIW).

On Tue, May 31, 2016 at 9:10 AM, Dale John McGrew <notifications@github.com>
wrote:

> In src/js/components/Widgets/PositionSupportOpposeSnippet.jsx
> <https://github.com/wevote/WebApp/pull/288#discussion_r65213527>:
>
> > +      className = "explicit-position__icon";
> > +      alt = "Supports";
> > +      positionLabel = "Supports";
> > +      isSupportedBy = "is Supported by"
> > +    } else if (this.props.is_oppose) {
> > +      src = "/img/global/icons/thumbs-down-color-icon.svg";
> > +      className = "explicit-position__icon";
> > +      alt = "Opposes";
> > +      positionLabel = "Opposes";
> > +      isSupportedBy = "is Opposed by"
> > +    } else {
> > +      src = "/img/global/icons/mixed-rating-icon.svg";
> > +      className = "position-rating__icon position-rating__icon--mixed";
> > +      alt = "Neutral Rating";
> > +      positionLabel = "Neutral Rating of";
> > +      isSupportedBy = "is viewed Neutrally by"
>
> Hi @ixley <https://github.com/ixley>, an example of a neutral
> "information only" rating would be KQED (or other nonpartisan organization)
> providing analysis, but no endorsement or opposition.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wevote/WebApp/pull/288/files/bd27987d1d712cbb8e6daf46ff7acbdcdd7292c3#r65213527>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAJzlvWdKJwATH26fOJ4LBvYdHpaFKFYks5qHF2MgaJpZM4Io8gk>
> .
>